lib/cli-engine.js

```diff
@@ -116,6 +116,66 @@ function calculateStatsPerRun(results) {
 }
 
 /**
+ * Performs multiple autofix passes over the text until as many fixes as possible
+ * have been applied.
+ * @param {string} text The source text to apply fixes to.
+ * @param {Object} config The ESLint config object to use.
+ * @param {Object} options The ESLint options object to use.
+ * @param {string} options.filename The filename from which the text was read.
+ * @param {boolean} options.allowInlineConfig Flag indicating if inline comments
+ *      should be allowed.
+ * @returns {Object} The result of the fix operation as returned from the
+ *      SourceCodeFixer.
+ * @private
+ */
+function multipassFix(text, config, options) {
+
+    var messages = [],
+        fixedResult,
+        fixed = false,
+        passNumber = 0,
+        lastMessageCount;
+
+    /**
+     * This loop continues until one of the following is true:
+     *
+     * 1. No more fixes have been applied.
+     * 2. There are no more linting errors reported.
```

Comment:
Hmm that's a good point. It does seem we always need at least one more pass whenever a fix was done.